src/backend/distributed/commands/vacuum.c

```diff
@@ -73,13 +79,68 @@ static CitusVacuumParams VacuumStmtParams(VacuumStmt *vacstmt);
 void
 PostprocessVacuumStmt(VacuumStmt *vacuumStmt, const char *vacuumCommand)
 {
-	int relationIndex = 0;
-	List *vacuumRelationList = ExtractVacuumTargetRels(vacuumStmt);
-	List *relationIdList = NIL;
 	CitusVacuumParams vacuumParams = VacuumStmtParams(vacuumStmt);
+
+	List *relationIdList = VacuumRelationIdList(vacuumStmt, vacuumParams);
+
+	if (ContainsOnlyPostgresTables(relationIdList))
```

Comment:
Perhaps `VacuumRelationIdList` can return only distributed tables (after a rename) to avoid this extra function.